File: languages/csharp/AppEncryption/AppEncryption/AppEncryptionPartition.cs
namespace GoDaddy.Asherah.AppEncryption
{
    public class AppEncryptionPartition
    {
        public AppEncryptionPartition(string partitionId, string systemId, string productId)
        {
            PartitionId = partitionId;
            SystemId = systemId;
            ProductId = productId;
        }

        public string SystemKeyId => "_SK_" + SystemId + "_" + ProductId;

        public string IntermediateKeyId => "_IK_" + PartitionId + "_" + SystemId + "_" + ProductId;

        internal string PartitionId { get; }

        internal string SystemId { get; }

        internal string ProductId { get; }

        public override string ToString()
        {
            return GetType().Name + "[partitionId=" + PartitionId +
                   ", systemId=" + SystemId + ", productId=" + ProductId + "]";
        }
    }
}

File: languages/java/app-encryption/src/main/java/com/godaddy/asherah/appencryption/AppEncryptionPartition.java
package com.godaddy.asherah.appencryption;

public class AppEncryptionPartition {

  private final String partitionId;
  private final String systemId;
  private final String productId;

  public AppEncryptionPartition(final String partitionId, final String systemId, final String productId) {
    this.partitionId = partitionId;
    this.systemId = systemId;
    this.productId = productId;
  }

  String getPartitionId() {
    return partitionId;
  }

  String getSystemId() {
    return systemId;
  }

  String getProductId() {
    return productId;
  }

  public String getSystemKeyId() {
    return "_SK_" + systemId + "_" + productId;
  }

  public String getIntermediateKeyId() {
    return "_IK_" + partitionId + "_" + systemId + "_" + productId;
  }

  @Override
  public String toString() {
    return getClass().getSimpleName() + "[partitionId=" + partitionId +
      ", systemId=" + systemId + ", productId=" + productId + "]";
  }
}
